ci-operator/step-registry/openshift/image-based/upgrade/metal/config/openshift-image-based-upgrade-metal-config-commands.sh

```diff
@@ -31,6 +31,22 @@ cat <<EOF > ${SHARED_DIR}/install.sh
 #!/bin/bash
 set -euo pipefail
 
+function dnf_install_retry {
+  apps=(\$@)
+  echo "Apps I want to install are: \${apps[*]}"
+
+  for _ in \$(seq 3) ; do
+    sudo dnf clean -y all # clean the cache
+
+    # shellcheck disable=SC2086
+    sudo dnf install -y \${apps[*]} && return 0
+     
+    rc=\$? # save the return code
```

Comment:
Completely up to you if you think it's helpful, but it might be useful to have a log record that the loop executed, something like this might be useful?

```
if [ $rc -ne 0 ]
then
    echo "Failed to run dnf install, retrying"
fi
```